x/stakeibc/keeper/keeper.go

```diff
@@ -316,3 +320,14 @@ func (k Keeper) ConfirmValSetHasSpace(ctx sdk.Context, validators []*types.Valid
 
 	return nil
 }
+
+func (k msgServer) RegisterInterchainAccount(ctx sdk.Context, connectionId string, owner string, appVersion string) error {
```

Comment:
What is this change?

Reply:
the RegisterInterchainAccount API of the ICAcontrollerkeeper is deprecated in v7 so i have to use ICAcontrollerkeeper MsgServer to use the register interchain account service. And there is a lot of code that use the old RegisterInterchainAccount API so moving to a seperate function for msgserver logic remove duplicate code not